src/nvim/tui/termkey_defs.h

```diff
@@ -0,0 +1,228 @@
+  TERMKEY_FLAG_NOTERMIOS   = 1 << 4,  // Do not make initial termios calls on construction
+  TERMKEY_FLAG_SPACESYMBOL = 1 << 5,  // Sets TERMKEY_CANON_SPACESYMBOL
+  TERMKEY_FLAG_CTRLC       = 1 << 6,  // Allow Ctrl-C to be read as normal, disabling SIGINT
+  TERMKEY_FLAG_EINTR       = 1 << 7,  // Return ERROR on signal (EINTR) rather than retry
+  TERMKEY_FLAG_NOSTART     = 1 << 8,  // Do not call termkey_start() in constructor
+};
+
+enum {
+  TERMKEY_CANON_SPACESYMBOL = 1 << 0,  // Space is symbolic rather than Unicode
+  TERMKEY_CANON_DELBS       = 1 << 1,  // Del is converted to Backspace
+};
+
+typedef struct {
+  TermKeyType type;
+  union {
+    long codepoint;  // TERMKEY_TYPE_UNICODE
+    int number;    // TERMKEY_TYPE_FUNCTION
+    TermKeySym sym;       // TERMKEY_TYPE_KEYSYM
+    char mouse[4];  // TERMKEY_TYPE_MOUSE
+                    // opaque. see termkey_interpret_mouse
+  } code;
+
+  int modifiers;
+
+  // Any Unicode character can be UTF-8 encoded in no more than 6 bytes, plus
+  // terminating NUL
+  char utf8[7];
+} TermKeyKey;
+
+// Mostly-undocumented hooks for doing evil evil things
+typedef const char *TermKey_Terminfo_Getstr_Hook(const char *name, const char *value, void *data);
+
+typedef enum {
+  TERMKEY_FORMAT_LONGMOD     = 1 << 0,  // Shift-... instead of S-...
+  TERMKEY_FORMAT_CARETCTRL   = 1 << 1,  // ^X instead of C-X
+  TERMKEY_FORMAT_ALTISMETA   = 1 << 2,  // Meta- or M- instead of Alt- or A-
+  TERMKEY_FORMAT_WRAPBRACKET = 1 << 3,  // Wrap special keys in brackets like <Escape>
+  TERMKEY_FORMAT_SPACEMOD    = 1 << 4,  // M Foo instead of M-Foo
+  TERMKEY_FORMAT_LOWERMOD    = 1 << 5,  // meta or m instead of Meta or M
+  TERMKEY_FORMAT_LOWERSPACE  = 1 << 6,  // page down instead of PageDown
+
+  TERMKEY_FORMAT_MOUSE_POS   = 1 << 8,  // Include mouse position if relevant; @ col,line
+} TermKeyFormat;
+
+// Some useful combinations
+
+#define TERMKEY_FORMAT_VIM (TermKeyFormat)(TERMKEY_FORMAT_ALTISMETA|TERMKEY_FORMAT_WRAPBRACKET)
+
+typedef enum {
+  kKeyEncodingLegacy,  ///< Legacy key encoding
+  kKeyEncodingKitty,   ///< Kitty keyboard protocol encoding
+  kKeyEncodingXterm,   ///< Xterm's modifyOtherKeys encoding (XTMODKEYS)
+} KeyEncoding;
+
+#define KEY_BUFFER_SIZE 0x1000
+typedef struct {
+  int in_fd;
+  // Phases: -1=all 0=disabled 1=first-chunk 2=continue 3=last-chunk
+  int8_t paste;
+  bool ttimeout;
+
+  bool waiting_for_kkp_response;  ///< True if we are expecting to receive a response to a query for
+                                  ///< Kitty keyboard protocol support
+
+  KeyEncoding key_encoding;       ///< The key encoding used by the terminal emulator
+
+  OptInt ttimeoutlen;
+  TermKey *tk;
+  TermKey_Terminfo_Getstr_Hook *tk_ti_hook_fn;  ///< libtermkey terminfo hook
+  uv_timer_t timer_handle;
+  Loop *loop;
+  RStream read_stream;
+  TUIData *tui_data;
+  char key_buffer[KEY_BUFFER_SIZE];
+  size_t key_buffer_len;
+} TermInput;
```

Comment:
Why are `KeyEncoding` and `TermInput` moved to this file?

Reply:
Fix clint warnings. I can revert and include termkey_defs instead if you prefer

Reply:
Actually keyencoding has to be there to silence clint warning. So this is correct.

Reply:
What clint warning? There are no clint warnings on [`c180eef` (#30047)](https://github.com/neovim/neovim/pull/30047/commits/c180eef87187ec5b1b85b8bc831450d3665b8f1a)

Reply:
Why? I don't see it used in any other header.

Reply:
Must've looked at the wrong place or something. Fixed.